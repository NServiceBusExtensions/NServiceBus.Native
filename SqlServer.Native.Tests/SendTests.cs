using System;
using System.Collections.Generic;
using System.Text;
using ObjectApproval;
using SqlServer.Native;
using Xunit;

public class SendTests
{
    static DateTime dateTime = new DateTime(2000, 1, 1, 1, 1, 1, DateTimeKind.Utc);

    static SendTests()
    {
        DbSetup.Setup();
    }

    [Fact]
    public void SendSingle()
    {
        MessageQueueCreator.Drop(Connection.ConnectionString, "SendTests").Await();
        MessageQueueCreator.Create(Connection.ConnectionString, "SendTests").Await();
        var sender = new Sender();

        var message = BuildMessage("00000000-0000-0000-0000-000000000001");
        sender.Send(Connection.ConnectionString, "SendTests", message).Await();
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData("SendTests"));
    }

    [Fact]
    public void SendBatch()
    {
        MessageQueueCreator.Drop(Connection.ConnectionString, "SendTests").Await();
        MessageQueueCreator.Create(Connection.ConnectionString, "SendTests").Await();
        var sender = new Sender();

        sender.Send(
            Connection.ConnectionString, "SendTests",
            new List<OutgoingMessage>
            {
                BuildMessage("00000000-0000-0000-0000-000000000001"),
                BuildMessage("00000000-0000-0000-0000-000000000002")
            }).Await();
        ObjectApprover.VerifyWithJson(SqlHelper.ReadData("SendTests"));
    }

    static OutgoingMessage BuildMessage(string guid)
    {
        var headers = new Dictionary<string, string>
        {
            {"headerKey1", "headerValue1"},
            {"headerKey2", "headerValue2"}
        };

        return new OutgoingMessage(new Guid(guid), "theCorrelationId", "theReplyToAddress", dateTime, headers, Encoding.UTF8.GetBytes("{}"));
    }
}