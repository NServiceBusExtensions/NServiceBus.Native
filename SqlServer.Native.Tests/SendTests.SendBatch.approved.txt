[
  {
    "Id": "00000000-0000-0000-0000-000000000001",
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Recoverable": true,
    "Expires": "2000-01-01T01:01:01",
    "Headers": "{\"headerKey1\":\"headerValue1\",\"headerKey2\":\"headerValue2\"}",
    "BodyString": "絻",
    "Body": "e30=",
    "RowVersion": 1
  },
  {
    "Id": "00000000-0000-0000-0000-000000000002",
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Recoverable": true,
    "Expires": "2000-01-01T01:01:01",
    "Headers": "{\"headerKey1\":\"headerValue1\",\"headerKey2\":\"headerValue2\"}",
    "BodyString": "絻",
    "Body": "e30=",
    "RowVersion": 2
  }
]